### PR TITLE
[BACKLOG-7410] - Fatal Error occurs on attempt to press Cancel button in progress bar

### PR DESCRIPTION
**Pull request by prakapenka**
- do not erase report 'CANCELED' status in case of Exception

### CODE REVIEW

**Comment by prakapenka** [comment]
@tmorgner please review. Tag!

File: src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java [hunk 1]
```diff
@@ -18,13 +19,13 @@
...
-  private final SimpleReportingComponent reportComponent;
-  private final AsyncJobFileStagingHandler handler;
-  private final String url;
-  private final String auditId;
-  private final IPentahoSession safeSession;
+  protected final SimpleReportingComponent reportComponent;
+  protected final AsyncJobFileStagingHandler handler;
+  protected final String url;
+  protected final String auditId;
+  protected final IPentahoSession safeSession;
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java]: There is no need to set these variables as protected. Good encapsulation rules say that all data should be as invisible as possible. In the scope of this bug fix, there is no reason for this change.

File: src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
  * Simple synchronized bean with async execution status. This class is part of the PentahoAsyncReportExecution
  * implementation and should not be used outside of that.
  */
-class AsyncReportStatusListener implements IAsyncReportListener {
+public class AsyncReportStatusListener implements IAsyncReportListener {
```
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java]: The AsyncReportStatusListener is an implementation details of the AsyncReportExecution and is not useful outside of that context. There is no need to make that class public.

File: src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java (hunk 1 above)
Comment (prakapenka) [inline src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java]: There is a need to set these variables as protected since they are going to be re-used in PIR plugin. As far as making listener public. Since PIR don't follow reporting package structure, but has it as dependency.

File: src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java (hunk 2 above)
Comment (prakapenka) [inline src/org/pentaho/reporting/platform/plugin/async/AsyncReportStatusListener.java]: As far as we a going to have async mode for PIR....

**Comment by prakapenka** [comment]
@tmorgner please review, thank you in advance.

File: src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java (hunk 1 above)
Comment (tmorgner) [inline src/org/pentaho/reporting/platform/plugin/async/PentahoAsyncReportExecution.java]: If you are reusing them there, make that in a separate commit related to that case. Dont wildly mix work on two cases - if you work on two cases at the same time, use separate branches to keep that work cleanly separate.

**Comment by prakapenka** [comment]
@tmorgner please review
